### PR TITLE
feature/laf-4-item-new-version

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  18.4  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  7+7+1+1  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Problem description
Check the issue.

### TODO
- [x] Add the metadata: `dc.relation.replaces` and `dc.relation.isreplacedby` to the `local-types.xml`
- [x] Copy `VersionedHandleIdentifierProvider.java` and `VersionedDOIIdentifierProvider.java` and create `ClarinVersionedHandleIdentifierProvider` and `ClarinVersionedDOIIdentifierProvider`
- [x] When is creating the new version of the item: the previous item handle should be added to the metadata `dc.relation.replaces` of the new item, and the new item handle should be added to the metadata `dc.relation.isreplacedby` of the previous item.
- [x] Remove the code where is creating the handle based on the history.
- [x] Create Integration Tests

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/identifier/ClarinVersionedDOIIdentifierProvider.java [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    protected static final String pattern = "\\d+\\" + String.valueOf(DOT) + "\\d+";
+
+    @Autowired(required = true)
+    protected VersioningService versioningService;
+    @Autowired(required = true)
+    protected VersionHistoryService versionHistoryService;
+
+    @Override
+    public String mint(Context context, DSpaceObject dso)
+            throws IdentifierException {
```
Comment (mrovnanik) [inline dspace-api/src/main/java/org/dspace/identifier/ClarinVersionedDOIIdentifierProvider.java:56]: quite complicated. no chance to make it more readable?

File: dspace-api/src/main/resources/spring/spring-dspace-core-services.xml [hunk 2]
```diff
@@ -31,8 +31,14 @@
         <property name="configurationService" ref="org.dspace.services.ConfigurationService"/>
     </bean-->
 
-    <bean id="org.dspace.identifier.VersionedHandleIdentifierProvider"
-          class="org.dspace.identifier.VersionedHandleIdentifierProvider"
+<!--    <bean id="org.dspace.identifier.VersionedHandleIdentifierProvider"-->
```
Comment (mrovnanik) [inline dspace-api/src/main/resources/spring/spring-dspace-core-services.xml:34]: please remove or leave a comment, why it's necessary to have it commented out

**Review by mrovnanik — COMMENTED** [review]
just a small commentary

File: dspace-api/src/main/java/org/dspace/identifier/ClarinVersionedDOIIdentifierProvider.java (hunk 1 above)
Comment (milanmajchrak) [reply dspace-api/src/main/java/org/dspace/identifier/ClarinVersionedDOIIdentifierProvider.java:56]: This class is copied from the VersionedDOIIdentifierProvider. I've updated just a few rows. This mint method is complicated for me too but I wouldn't like to change it if it is not necessary.
I copied it to another class for better maintenance in the future .

File: dspace-api/src/main/resources/spring/spring-dspace-core-services.xml (hunk 2 above)
Comment (milanmajchrak) [reply dspace-api/src/main/resources/spring/spring-dspace-core-services.xml:35]: Added the comments why I commented it out, check the updated code: https://github.com/dataquest-dev/DSpace/blob/feature/laf-4-item-new-version/dspace-api/src/main/resources/spring/spring-dspace-core-services.xml#L34

**Pull request merged**